src/main/java/com/bina/varsim/tools/simulation/VCF2diploid.java

```diff
@@ -340,7 +341,7 @@ public void makeDiploid() {
      * @param InsertPosition2Sequence --  records insertion locations and sequences
      * @return true if the variant is incorporated, false otherwise
      */
-    private boolean addVariant(byte[] maskedSequence, Sequence referenceSequence,
+    private boolean addVariant(byte[] maskedSequence, Sequence referenceSequence, SimpleReference allSequences,
```

Comment:
Please use the `final` keyword liberally for variables/parameters to show intent that they are not expected to be modified once set.